packages/scene-composer/src/hooks/useMatterportTags.ts

```diff
@@ -164,31 +165,61 @@ const updateTag = async (
       return;
     }
 
-    const tagComp = tagIndex !== -1 ? updateTagEntityComponent(components[tagIndex]) : undefined;
+    const isUpdate = !!node.properties.layerIds?.at(0);
+    const tagComp =
+      tagIndex !== -1
+        ? (isUpdate ? updateTagEntityComponent : createTagEntityComponent)(components[tagIndex])
+        : undefined;
     const overlayComp =
       dataOverlayIndex !== -1
-        ? updateOverlayEntityComponent(components[dataOverlayIndex] as IDataOverlayComponent)
+        ? (isUpdate ? updateOverlayEntityComponent : createOverlayEntityComponent)(
+            components[dataOverlayIndex] as IDataOverlayComponent,
+          )
         : undefined;
-    const nodeComp = updateNodeEntityComponent(node, layerId);
+    const nodeComp = (isUpdate ? updateNodeEntityComponent : createNodeEntityComponent)(node, layerId);
 
-    const updateEntity: UpdateEntityCommandInput = {
-      workspaceId: undefined,
-      entityId: ref,
-      entityName: item.label + '_' + ref,
-      componentUpdates: {
-        Node: nodeComp,
-      },
-    };
-    if (tagComp) {
-      updateEntity.componentUpdates![KnownComponentType.Tag] = tagComp;
-    }
-    if (overlayComp) {
-      updateEntity.componentUpdates![KnownComponentType.DataOverlay] = overlayComp;
-    }
-    try {
-      await sceneMetadataModule.updateSceneEntity(updateEntity);
-    } catch (e) {
-      console.error('Update scene node entity failed', e);
+    if (isUpdate) {
+      const updateEntity: UpdateEntityCommandInput = {
+        workspaceId: undefined,
+        entityId: ref,
+        entityName: item.label + '_' + ref,
+        componentUpdates: {
+          Node: nodeComp,
+        },
+      };
+      if (tagComp) {
+        updateEntity.componentUpdates![KnownComponentType.Tag] = tagComp;
+      }
+      if (overlayComp) {
+        updateEntity.componentUpdates![KnownComponentType.DataOverlay] = overlayComp;
+      }
+
+      try {
+        await sceneMetadataModule.updateSceneEntity(updateEntity);
+      } catch (e) {
+        console.error('Update scene node entity failed', e);
+      }
+    } else {
+      const createEntity: CreateEntityCommandInput = {
```

Comment:
Looks like there is no difference in parameters except ParentEntityId. You can pass them as parameter and have them in a single request? or add `parentEntityId` based on the request?

Reply:
I tried but make code less readable. the API called is different, input type is different, one use `components` and other use `componentUpdates`, and one with `parentEntityId`

Reply:
create entity has `components` instead of `componentUpdates`, and we cannot assign `undefined` value to them.
create has `parentEntityId` but update doesn't.
having the API call here is to minimize the impact of the dynamic scene change. for R2 when we support other kind of nodes, I will refactor this part to be consistent with other nodes